### PR TITLE
Improved SplitTurretAimAnimation update rule

### PR DESCRIPTION
**Pull request by reaperrr**
- Now lists each updated location after update
- Now properly handles the unlikely-but-possible case of "only ReloadPrefix, no Sequences"
- Dropped load-time exception when an actor has both `WithTurretAimAnimation` and `WithTurretAttackAnimation`. While they don't mesh very well, forcing modders to not use them together when it's rather a design issue/bug than something really intentional is a bit harsh. Also, I'm working on a fix already.

Split from #15140.

### CODE REVIEW

File: OpenRA.Mods.Common/UpdateRules/Rules/SplitTurretAimAnimation.cs [hunk 1]
```diff
@@ -26,33 +27,54 @@ public override string Description
 			}
 		}
 
+		readonly List<Tuple<string, string>> locations = new List<Tuple<string, string>>();
+
+		public override IEnumerable<string> AfterUpdate(ModData modData)
+		{
+			var message = "AimSequence and ReloadPrefix have been split from WithTurretAttackAnimation\n"
+				+ "to the new WithTurretAimAnimation trait. The following location has been updated,\n"
```
Comment (pchote) [inline OpenRA.Mods.Common/UpdateRules/Rules/SplitTurretAimAnimation.cs]: "location has" &rarr; "locations have"

**Comment by pchote** [comment]
This appears to have broken the rule (or perhaps it was already broken?).

Repro case:
1. check out release-20180307 and make a copy of the cnc mod.
2. Update mod.yaml to change $cnc to the new folder name.
3. Switch to this branch and compile the project.
4. Run `--update-mod SplitTurretAimAnimation`.

Expected result: the same yaml changes that were made in #14846.
Actual result: The yaml was not updated.

File: OpenRA.Mods.Common/UpdateRules/Rules/SplitTurretAimAnimation.cs [hunk 2]
```diff
@@ -26,33 +27,54 @@ public override string Description
...
+				+ "you may want to check the result for any necessary manual adjustments:\n"
+				+ UpdateUtils.FormatMessageList(locations.Select(n => n.Item1 + " (" + n.Item2 + ")"));
+
+			if (locations.Any())
+				yield return message;
```
Comment (pchote) [inline OpenRA.Mods.Common/UpdateRules/Rules/SplitTurretAimAnimation.cs]: This doesn't seem to work.  If I revert the yaml change from #14846 and run the update rule it splits off `WithTurretAimAnimation` but does not report any locations.

**Comment by reaperrr** [comment]
Updated, locations include (former) WithSpriteTurret.AimSequence locations now. Also rewrote the update message to account for that.

**Comment by abcdefg30** [comment]
> OpenRA.Mods.Common/UpdateRules/Rules/SplitTurretAimAnimation.cs:L104: [SP2000] Invalid spacing at the end of the line.

**Comment by reaperrr** [comment]
Fixed.

**Comment by pchote** [comment]
I'm quite confused about the `ReloadPrefix` changes here, and with the situation in general between this and #15140.  Can we please merge things back into a single PR if that is what is needed to keep related changes together?

**Comment by reaperrr** [comment]
Closing, fused the parts that make sense back into #15140's update rule.

**Pull request closed without merging**